pkg/policy/execute.go

```diff
@@ -102,11 +105,33 @@ func NewExecutor(conn *pgxpool.Conn, log hclog.Logger, progressUpdate UpdateCall
 		conn:           conn,
 		log:            log,
 		progressUpdate: progressUpdate,
+		PolicyPath:     &PolicyPath{policyPath: []string{}},
 	}
 }
 
+type PolicyPath struct {
+	policyPath []string
+}
+
+func (e *PolicyPath) append(policyPath string) {
+	e.policyPath = append(e.policyPath, policyPath)
+}
+
+func (e *PolicyPath) pop() {
+	e.policyPath = e.policyPath[:len(e.policyPath)-1]
+}
+
+func (e *Executor) with(args ...interface{}) {
+	e.log = e.log.With(args...)
+}
+
+func (e *Executor) withPolicyPath() {
+	e.log = e.log.With("policy", strings.Join(e.PolicyPath.policyPath, "/"))
+}
+
```

Comment:
You can do all the joining logic here With should return a *NEW* executor instance similar to how e.log.With works i.e e.log != e.log.With. This will remove the need for pop and every time you do With("policyName") -> *Executor instance

```suggestion
func (e *Executor) with(policy string, args ...interface{}) *Executor {
	return &Executor{
	 conn: e.conn,
	 e.log: e.log.With("policy", strings.Join(e.PolicyPath.policyPath, "/"), args...)
	 progressUpdate: e.progressUpdate
	}
}

```